gitlab/resource_gitlab_project.go

```diff
@@ -915,129 +915,142 @@ func resourceGitlabProjectDelete(ctx context.Context, d *schema.ResourceData, me
 func editOrAddPushRules(ctx context.Context, client *gitlab.Client, projectID string, d *schema.ResourceData) error {
 	log.Printf("[DEBUG] Editing push rules for project %q", projectID)
 
-	editOptions := expandEditProjectPushRuleOptions(d)
-	_, _, err := client.Projects.EditProjectPushRule(projectID, editOptions, gitlab.WithContext(ctx))
-	if err == nil {
-		return nil
-	}
+	pushRules, _, err := client.Projects.GetProjectPushRules(d.Id(), gitlab.WithContext(ctx))
+	// NOTE: push rules id `0` indicates that there haven't been any push rules set.
+	if err != nil || is404(err) || pushRules.ID == 0 {
```

Comment:
It still looks like you are ignoring any errors from calling GetProjectPushRules, like an HTTP 500 error.

Reply:
Yes, actually by intention - I thought for an error message to the user it might be more intuitive to have the subsequent `AddProjectPushRule` fail, because that was what the user tried to achieve: set the push rules. 

But, I see your point and can easily change it. What's our general approach here? 🤔

Reply:
I see what you mean. That's all right, as long as continuing after an unexpected error won't harm anything. In that case the `is404(err)` check is redundant.